mender-update/http_resumer/http_resumer.cpp

```diff
@@ -265,16 +277,12 @@ void BodyHandlerFunctor::operator()(http::ExpectedIncomingResponsePtr exp_resp)
 		resumer_client->response_->headers_ = resp->GetHeaders();
 
 		// Finished, call the user handler \o/
-		resumer_client->Disable();
-		resumer_client->CallUserHandler(resumer_client->response_);
 		resumer_client->logger_.Debug("Download resumed and completed successfully");
+		resumer_client->DoCancel();
```

Comment:
I don't think you need this. The connection is already in a cancelled state after a successful handler call.

Reply:
Oops, my mistake, I misread this. I though you were calling it on the real client. Never mind this comment.

Reply:
Actually, this one I need.

Note that it is calling `DoCancel` (and not `Cancel`) to set our `cancelled_` pointer to true. Without this the client object is not reusable and subsequent calls to `AsyncCall` will error with `HTTP call already ongoing`

Reply:
Yep, looks correct to me!